src/common/I18n.js

```diff
@@ -6,7 +6,7 @@ class I18n {
    * Constructor.
    *
    * @param {Object} options Options.
-   * @param {string} options.locale Locale.
+   * @param {string=} options.locale Locale.
```

Comment:
@qwerty541 I now see that the default value is hidden inside the constructor. Maybe we can replace it with the following:

```js
constructor({ locale = FALLBACK_LOCALE, translations }) {
    this.locale = locale;
    this.translations = translations;
}
```

Reply:
Additionally, the `t` method could be improved:

```js
t(str) {
    const { translations, locale } = this;
    const translation = translations[str]?.[locale];
    if (!translation) {
        throw new Error(`'${str}' translation not found for locale '${locale}'`);
    }
    return translation;
}
```

Reply:
@rickstaa This pull request resolves the following vscode type errors:

src/cards/repo-card.js on line 94
src/cards/stats-card.js on line 150
src/cards/top-languages-card.js on line 744
src/cards/wakatime-card.js on line 208

This occurs because `locale` is optional query parameter and can be `undefined` and I18n class constructor jsdoc requires string.

Reply:
> Additionally, the `t` method could be improved:
> 
> ```js
> t(str) {
>     const { translations, locale } = this;
>     const translation = translations[str]?.[locale];
>     if (!translation) {
>         throw new Error(`'${str}' translation not found for locale '${locale}'`);
>     }
>     return translation;
> }
> ```

I like current variant of this methos more because it contains different error messages when locale and translation is missing.

Reply:
> > Additionally, the `t` method could be improved:
> > ```js
> > t(str) {
> >     const { translations, locale } = this;
> >     const translation = translations[str]?.[locale];
> >     if (!translation) {
> >         throw new Error(`'${str}' translation not found for locale '${locale}'`);
> >     }
> >     return translation;
> > }
> > ```
> 
> I like current variant of this methos more because it contains different error messages when locale and translation is missing.

No problem, both are good! How about moving the fallback into the object deconstruction 🤔?

Reply:
Meaning:

```js
constructor({ locale = FALLBACK_LOCALE, translations }) {
    this.locale = locale;
    this.translations = translations;
}
```

Reply:
> > > Additionally, the `t` method could be improved:
> > > ```js
> > > t(str) {
> > >     const { translations, locale } = this;
> > >     const translation = translations[str]?.[locale];
> > >     if (!translation) {
> > >         throw new Error(`'${str}' translation not found for locale '${locale}'`);
> > >     }
> > >     return translation;
> > > }
> > > ```
> > 
> > 
> > I like current variant of this methos more because it contains different error messages when locale and translation is missing.
> 
> No problem, both are good! How about moving the fallback into the object deconstruction 🤔?

Moving the fallback into the object deconstruction looks good for me, but i have doubts about if it should be done inside this pull request or inside a new one, since this one about fixing jsdoc.

Reply:
Also thank you for following, i appreciate it, followed you back.

Reply:
> Moving the fallback into the object deconstruction looks good for me, but i have doubts about if it should be done inside this pull request or inside a new one, since this one about fixing jsdoc.

Let's do this in a new PR.

> Also thank you for following, i appreciate it, followed you back.

Of course. I'm curious about your projects and contributions 🚀!

Reply:
Recently i tried to make change with moving the fallback into the object deconstruction. Looks like this does not suit us because when `locale` variable passed into constructor with `undefined` value it override the default value bacause `undefined` is valid. Currently it will be handled by the following code `this.locale = locale || FALLBACK_LOCALE;`. 

Moving fallback can de done only the following way:

```js
constructor({ locale = FALLBACK_LOCALE, translations }) {
    // Check if 'locale' is undefined and set it to the default value if true
    if (locale === undefined) {
      locale = FALLBACK_LOCALE;
    }

    this.locale = locale;
    this.translations = translations;
  }
```

But i do not think that code will be better after this.

Reply:
I agree. Thanks for testing my improvement. I forgot about the fact that undefined is seen as a valid input! I think we can leave the old code 👍🏻.